pkg/restapi/issuer/operation/operations.go

```diff
@@ -519,12 +520,29 @@ func (o *Operation) createIssuerProfile(pr *ProfileRequest) (*vcprofile.DataProf
 
 	created := time.Now().UTC()
 
+	edvVaultID, err := o.createIssuerProfileVault(pr.Name)
+	if err != nil {
+		return nil, err
+	}
+
 	return &vcprofile.DataProfile{Name: pr.Name, URI: pr.URI, Created: &created, DID: didID,
-		SignatureType: pr.SignatureType, SignatureRepresentation: pr.SignatureRepresentation, Creator: publicKeyID,
-		DisableVCStatus: pr.DisableVCStatus, OverwriteIssuer: pr.OverwriteIssuer,
+		SignatureType: pr.SignatureType, SignatureRepresentation: pr.SignatureRepresentation, EDVVaultID: edvVaultID,
+		Creator: publicKeyID, DisableVCStatus: pr.DisableVCStatus, OverwriteIssuer: pr.OverwriteIssuer,
 	}, nil
 }
 
+// createIssuerProfileVault creates the vault associated with the profile
+func (o *Operation) createIssuerProfileVault(profileName string) (string, error) {
+	vaultLocationURL, err := o.edvClient.CreateDataVault(&models.DataVaultConfiguration{ReferenceID: profileName})
+	if err != nil {
+		return "", err
```

Comment:
I suggest wrapping this error - it will make debugging any issues later easier

Reply:
(note that these "wrap error" comments apply to other places as well where you have unwrapped errors

Reply:
fixed!